### ISSUE TITLE
Fix false positives for mixin-as-function syntax in function-whitespace-after

### ISSUE THREAD

**Issue by thybzi** [stylelint/stylelint#4299]
> Clearly describe the bug

There are false positives when using LESS feature of "mixins as functions": http://lesscss.org/features/#mixins-feature-mixins-as-functions-feature

> Which rule, if any, is the bug related to?

`function-whitespace-after`
(we set it to "always")

> What CSS is needed to reproduce the bug?

e.g.

```less
div {
    width: .mixin(@var)[];
}
```

> What stylelint configuration is needed to reproduce the bug?

e.g.

```json
{
    "rules": {
        "color-hex-case": "lower",
        "color-hex-length": "short",
        "color-named": "never",
        "color-no-invalid-hex": true,
        "font-family-name-quotes": "always-where-recommended",
        "function-blacklist": ["expression"],
        "function-calc-no-unspaced-operator": true,
        "function-comma-space-after": "always-single-line",
        "function-comma-space-before": "never",
        "function-linear-gradient-no-nonstandard-direction": true,
        "function-max-empty-lines": 0,
        "function-name-case": "lower",
        "function-parentheses-space-inside": "never-single-line",
        "function-url-no-scheme-relative": true,
        "function-url-quotes": "never",
        "function-whitespace-after": "always",
        "number-leading-zero": "always",
        "number-no-trailing-zeros": true,
        "string-no-newline": true,
        "string-quotes": "single",
        "length-zero-no-unit": true,
        "unit-case": "lower",
        "unit-no-unknown": true,
        "unit-whitelist": ["px", "%", "em", "s", "ms", "deg", "grad", "turn", "vh", "vw"],
        "value-no-vendor-prefix": true,
        "value-list-comma-newline-after": "always-multi-line",
        "value-list-comma-newline-before": "never-multi-line",
        "value-list-comma-space-before": "never",
        "value-list-max-empty-lines": 0,
        "custom-property-empty-line-before": "never",
        "shorthand-property-no-redundant-values": true,
        "property-blacklist": ["box-sizing", {
            "message": "Unexpected \"box-sizing\" property, use .box-sizing() mixin"
        }],
        "property-case": "lower",
        "property-no-unknown": [true, {
            "checkPrefixed": false,
            "ignoreProperties": ["/./", "transform+_"]
        }],
        "property-no-vendor-prefix": true,
        "keyframe-declaration-no-important": true,
        "declaration-bang-space-after": "never",
        "declaration-bang-space-before": "always",
        "declaration-colon-newline-after": "always-multi-line",
        "declaration-colon-space-after": "always-single-line",
        "declaration-colon-space-before": "never",
        "declaration-property-unit-whitelist": {
            "transition": ["s", "ms"],
            "animation": ["s", "ms"]
        },
        "declaration-block-no-duplicate-properties": [true, {
            "ignore": ["consecutive-duplicates-with-different-values"]
        }],
        "declaration-block-no-shorthand-property-overrides": true,
        "declaration-block-semicolon-newline-after": "always",
        "declaration-block-semicolon-newline-before": "never-multi-line",
        "declaration-block-semicolon-space-after": "never-single-line",
        "declaration-block-semicolon-space-before": "never",
        "declaration-block-single-line-max-declarations": 1,
        "declaration-block-trailing-semicolon": "always",
        "block-closing-brace-empty-line-before": "never",
        "block-closing-brace-newline-after": ["always", {
            "ignoreAtRules": ["grid", "func"]
        }],
        "block-closing-brace-newline-before": "always",
        "block-opening-brace-newline-after": "always",
        "block-opening-brace-space-before": "always",
        "selector-attribute-brackets-space-inside": "never",
        "selector-attribute-operator-space-after": "never",
        "selector-attribute-operator-space-before": "never",
        "selector-attribute-quotes": "never",
        "selector-class-pattern": ["[a-z][a-z0-9_-]", {
            "resolveNestedSelectors": true
        }],
        "selector-combinator-space-after": "always",
        "selector-combinator-space-before": "always",
        "selector-descendant-combinator-no-non-space": true,
        "selector-max-id": 0,
        "selector-no-vendor-prefix": true,
        "selector-pseudo-class-case": "lower",
        "selector-pseudo-class-no-unknown": true,
        "selector-pseudo-class-parentheses-space-inside": "never",
        "selector-pseudo-element-case": "lower",
        "selector-pseudo-element-colon-notation": "double",
        "selector-pseudo-element-no-unknown": true,
        "selector-type-case": "lower",
        "selector-type-no-unknown": [true, {
            "ignoreTypes": ["when", "and"]
        }],
        "selector-max-empty-lines": 1,
        "selector-list-comma-newline-after": "always",
        "selector-list-comma-newline-before": "never-multi-line",
        "selector-list-comma-space-before": "never",
        "rule-empty-line-before": [ "always", {
            "ignore": ["after-comment"]
        }],
        "media-feature-colon-space-after": "always",
        "media-feature-colon-space-before": "never",
        "media-feature-name-case": "lower",
        "media-feature-name-no-unknown": true,
        "media-feature-parentheses-space-inside": "never",
        "media-feature-range-operator-space-after": "always",
        "media-feature-range-operator-space-before": "always",
        "media-query-list-comma-newline-after": "always",
        "media-query-list-comma-newline-before": "never-multi-line",
        "media-query-list-comma-space-after": "never-single-line",
        "media-query-list-comma-space-before": "never",
        "at-rule-name-case": "lower",
        "at-rule-name-space-after": "always-single-line",
        "at-rule-no-vendor-prefix": true,
        "at-rule-semicolon-newline-after": "always",
        "comment-no-empty": true,
        "comment-whitespace-inside": "always",
        "indentation": [4, {
            "ignore": ["inside-parens", "param"]
        }],
        "max-empty-lines": 3,
        "max-line-length": 160,
        "no-eol-whitespace": true,
        "no-missing-end-of-source-newline": true,
        "no-unknown-animations": true
    }
}
```

> Which version of stylelint are you using?

`11.0.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

grunt-stylelint 0.11.0 (which uses Node.js API)

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, it's related to LESS "mixins as functions" feature: http://lesscss.org/features/#mixins-feature-mixins-as-functions-feature

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

The following warnings were flagged:

```shell
Expected whitespace after ")" (function-whitespace-after)
```

**Comment by evilebottnawi** [comment]
We should skip non standard syntax

**Comment by jeddy3** [comment]
Labelling as "help wanted". Feel free to submit a PR.

**Comment by jeddy3** [comment]
Closing [as stylistic rules are frozen](https://github.com/stylelint/stylelint/issues/5674). The community is welcome to migrate the rule [to a plugin](https://stylelint.io/developer-guide/plugins) and fix this bug.
